variables-deprecated.tf

```diff
@@ -1,101 +1,103 @@
     EOF
+  default     = null
 }
 
 variable "lifecycle_prefix" {
   type        = string
   description = "(Deprecated, use `lifecycle_configuration_rules` instead)\nPrefix filter. Used to manage object lifecycle events"
-  default     = ""
+  default     = null
 }
 
 variable "lifecycle_tags" {
   type        = map(string)
   description = "(Deprecated, use `lifecycle_configuration_rules` instead)\nTags filter. Used to manage object lifecycle events"
-  default     = {}
+  default     = null
 }
 
 variable "abort_incomplete_multipart_upload_days" {
   type        = number
-  default     = 5
   description = "(Deprecated, use `lifecycle_configuration_rules` instead)\nMaximum time (in days) that you want to allow multipart uploads to remain in progress"
+  default     = null
 }
 
 variable "enable_glacier_transition" {
   type        = bool
-  default     = true
   description = "(Deprecated, use `lifecycle_configuration_rules` instead)\nEnables the transition to AWS Glacier which can cause unnecessary costs for huge amount of small files"
+  default     = null
 }
 
 variable "enable_noncurrent_version_expiration" {
   type        = bool
-  default     = true
   description = "(Deprecated, use `lifecycle_configuration_rules` instead)\nEnable expiration of non-current versions"
+  default     = null
 }
 
 variable "expiration_days" {
   type        = number
-  default     = 90
   description = "(Deprecated, use `lifecycle_configuration_rules` instead)\nNumber of days after which to expunge the objects"
+  default     = null
 }
 
 variable "standard_transition_days" {
   type        = number
-  default     = 30
   description = "(Deprecated, use `lifecycle_configuration_rules` instead)\nNumber of days to persist in the standard storage tier before moving to the infrequent access tier"
+  default     = null
 }
 
 variable "glacier_transition_days" {
   type        = number
-  default     = 60
   description = "(Deprecated, use `lifecycle_configuration_rules` instead)\nNumber of days after which to move the data to the Glacier Flexible Retrieval storage tier"
+  default     = null
 }
 
 variable "noncurrent_version_transition_days" {
   type        = number
-  default     = 30
   description = "(Deprecated, use `lifecycle_configuration_rules` instead)\nSpecifies (in days) when noncurrent object versions transition to Glacier Flexible Retrieval"
+  default     = null
 }
 
 variable "noncurrent_version_expiration_days" {
   type        = number
-  default     = 90
   description = "(Deprecated, use `lifecycle_configuration_rules` instead)\nSpecifies when non-current object versions expire (in days)"
+  default     = null
 }
 
 
 
 locals {
   deprecated_lifecycle_rule = {
-    enabled = var.lifecycle_rule_enabled
+    enabled = var.lifecycle_rule_enabled == true || (var.lifecycle_rule_enabled == null && length(var.lifecycle_configuration_rules) == 0)
```

Comment:
```suggestion
    enabled = var.lifecycle_rule_enabled || (var.lifecycle_rule_enabled == null && length(var.lifecycle_configuration_rules) == 0)
```

Reply:
@nitrocode 
```
│ Error: Operation failed
│ 
│   on variables-deprecated.tf line 82, in locals:
│    82:   enabled = var.lifecycle_rule_enabled || (var.lifecycle_rule_enabled == null && length(var.lifecycle_configuration_rules) == 0)
│      ├────────────────
│      │ var.lifecycle_rule_enabled is null
│ 
│ Error during operation: argument must not be null.
```

Reply:
ah interesting, I didn't realize if a var `type = bool` that defaulted to `null` wouldn't already implicitly resolve to `false`. Good to know.